R/Data-methods.R

```diff
@@ -93,6 +93,119 @@ setMethod(
   }
 )
 
+## Data ----
+
+#' Plot Method for the [`Data`] Class
+#'
+#' A method that creates a profile plot for [`Data`] object.
+#'
+#' @param x (`Data`)\cr object we want to plot, which includes all evaluable subjects.
+#' @param x_NE A data.frame, which includes all non-evaluable subjects.
+#' @param unit A string specifying the dose unit used in the trials, for example "mg/kg".
+#'
+#' @return The [`ggplot2`] object.
+#'
+#' @examples
+#' \dontrun{
+#' dose_grid <- c(10, 15, 30, 45, 60, 80)
+#' unit <- "mg/kg"
+#' ## Specify the observed data
+#' data <- Data(
+#'   x = c(rep(10, 3), rep(15, 3), rep(30, 3)),
+#'   y = c(rep(0, 3), rep(0, 2), 1, rep(0, 2), 1),
+#'   cohort = c(rep(1, 3), rep(2, 3), rep(3, 3)),
+#'   doseGrid = dose_grid,
+#'   ID = 1:9
+#' )
+#' ## Specify the NON-evaluable data
+#' ## (if none, set data_NE <- data.frame(NULL))
+#' data_NE <- data.frame(
+#'   IDs = 10,
+#'   doses = 30,
+#'   dlts = 2,
+#'   cohorts = 3
+#' )
+#'
+#' plot(
+#'   x = data,
+#'   x_NE = data_NE,
+#'   unit = unit
+#' )
+#' }
+#' @source This function uses \code{ggplot} function from \code{ggplot2}
+#' R package.
+#' @export
+setGeneric(
+  name = "profiles",
+  def = function(x, xNE, unit, ...) {
+    standardGeneric("profiles")
+  }
+)
+setMethod(
+  f = "profiles",
+  signature = signature(x = "Data", xNE = "data.frame", unit = "character"),
+  definition = function(x, xNE, unit, ...) {
+    ID <- c(x@ID, xNE$IDs)
+    dose <- c(x@x, xNE$doses)
+    cohort <- c(x@cohort, xNE$cohorts)
+    DLT <- c(x@y, xNE$dlts)
+
+    df <- data.frame(
+      "ID" = ID,
+      "dose" = dose,
+      "DLT" = factor(DLT,
+        levels = c("0", "1", "2"),
+        labels = c("DLT No", "DLT Yes", "Not evaluable")
+      ),
+      "cohort" = paste("Cohort", cohort)
+    )
+    df <- df[order(df$ID), ]
+
+    p <- ggplot(data = df) +
+      geom_point(
+        aes(
+          x = factor(ID, levels = unique(ID[order(cohort, ID)])),
```

Comment:
I think we might need to work with 
```suggestion
          x = factor(.data$ID, levels = unique(.data$ID[order(.data$cohort, .data$ID)])),
```
etc. to avoid `R CMD check` notes about missing global variables, please double check the `R CMD check` output